### DISCUSSION TITLE
Hardware Requirement

### DISCUSSION THREAD

**Discussion by mishrankit9** [blakeblackshear/frigate#3056] [Ask A Question]
Hello All,
I wanted to configure 150 hikvision, all 5MP cameras into frigate.
The one given in the document says several cameras: https://docs.frigate.video/hardware/
What should be the recommended hardware?
Can I use one of the below?
1. Odyssey X86 Blue J4125 
2. Minisforum GK41 
3. Beelink GK55

Application:
-----------
1. Home-Assistant OS
2. Frigate
There will be few more applications which will be installed over the course of time, with regards to face-recognition, ANPR, etc.

My sincere request to the community, if someone has already tried and has some experience to share, I will be blessed.

**Comment by blakeblackshear** [comment]
Frigate has never been tested with that number of cameras, and I am not sure it would work. At the very least, you will need far more powerful hardware and multiple Coral devices. You may even need to consider multiple separate instances of frigate. The most I have seen someone use is ~35. Exactly what you need will depend on how much activity each camera has and what resolution you will use for object detection.

**Reply by mishrankit9** [reply]
Dear Sir,

Thanks for you quick response.

So lets say, I added 32 cameras under 1 frigate installation as maximum tried till now is 35.
- All Camera are 5MP resolution
- HomeAssistant OS installation
My current activities are
1. 24*7 recording
2. Event on motion (object tracking are people, car, bicycle, motorbike, bus
My Future activities will be
3. Install face recognition plugin like compreface or something similar
4. Install ANPR plugin like openalpr or something similar

Can you please suggest a hardware for the above configuration, or give some thumb of rule into deciding on the hardware, I will add some buffer before i purchase the hardware. In worst-case scenario, I will have to decrease the number of cameras from one hardware which is totally fine for me. I just want to start on to something reasonable.

**Reply by blakeblackshear** [reply]
Please read these parts of the documentation in detail:

https://docs.frigate.video/guides/camera_setup
https://docs.frigate.video/installation

- Home Assistant OS is not a recommended installation type
- You almost certainly won't want to run detection on the 5MP stream
- Monitoring a rural area where most cameras won't see objects often is completely different than cameras on a busy street
- I don't know anything about the resources to run 3 or 4

You could need anything from a dedicated server farm to a very powerful workstation for 32 cameras. Just the decoding alone will overwhelm the options you listed many many times over. I would see what hardware is recommended for large blue iris systems and use that as a starting point.

**Comment by PrplHaz4** [comment]
That is a huge deployment - the things I would be concerned about with this approach:
- lack of footage archiving/saved events and tiered storage
- UI not optimal for navigating what is sure to be a very large # of events
- number of coral devices or video cards required for object detection (this is an issue for whatever software you choose)
- effort to manage 5+ instances of HA (multiple updates every month?) and 5+ instances of Frigate

If you haven't done so you should try a smaller test deployment to get a feel for what your real needs/limitations are.

**Comment by jvrobert** [comment]
I would wonder at just the raw IO requirements for that much data, doesn't seem feasible on a single machine though I haven't done the math.
I mainly chimed in to say that if someone builds some sort of meta-frigate that lets you use multiple frigate instances from a nice single interface, it _must_ be named "Flotilla".
